### ISSUE TITLE
Poor handling of arguments in editor setting

### ISSUE THREAD

**Issue by Earnestly** [vimperator/vimperator-labs#333]
A few terminal emulators parse the defacto standard `-e` flag in a positional manner to let people use `terminal -e command --with --arguments` and correctly passing those arguments to the command instead of the terminal.

This creates a situation forces `-e` to be positional, no other terminal options can be set beyond this flag.  Some [terminals don't implement this behaviour](https://github.com/thestinger/termite/issues/246) for precisely this reason.

The issue is that in this case using `set editor="termite -e vim"` results in what I can only assume as conceptually `exec("termite", "-e", "vim", "temporary_file")` from [looking at vimperator's source](https://github.com/vimperator/vimperator-labs/blob/master/common/content/editor.js#L256)

The issue here is `"termporary_file"` is being passed as an argument to `termite` instead of `vim`. The solution given was to use quotes, which is simple enough when you're already using a shell but can become problematic in these kinds of situations.

Perhaps a simpler suggestion would be to expose a placeholder which is expanded as-is, such as `set editor='termite -e "vim <file>"'`.  Although there may still be a requirement for handling quotes in a shell-like manner.  

That said, implementing something like that might make it possible to use any command instead of assuming the behaviour of `-e` for some terminal emulators.

What do you think?

**Comment by bzub** [comment]
I have another workaround that will allow you to keep using Termite.

Create a wrapper script in your $PATH and make it executable.  In my case it's called termite-vim.  So I use `set editor=termite-ext` in my vimperatorrc.

Here's the wrapper script:
```
#!/bin/sh
termite -e "vim $1"
```

**Comment by Earnestly** [comment]
Yeah, that is an option, ideally I'd not want to use that though

**Comment by gkatsev** [comment]
Yeah, that sounds like a good addition. We should see what vim does for options like these. Is it `<file>`? Or would `%` work?
In the meantime, I think you'll need to use the workaround.
